solr/core/src/java/org/apache/solr/core/SolrCores.java

```diff
@@ -99,37 +85,34 @@ protected void close() {
     waitForLoadingCoresToFinish(30 * 1000);
     Collection<SolrCore> coreList = new ArrayList<>();
```

Comment:
what about moving the coreList definition inside the while block?

Reply:
I'd be happy to do that; I debated that but your suggestion now motivates me :-)